File: Backend/src/routes/auth.routes.js
import { Router } from 'express';
import { body } from 'express-validator';
import { register, login, logout, refreshToken, getMe } from '../controllers/auth.controller.js';
import { protect } from '../middlewares/auth.middleware.js';
import { validate } from '../middlewares/validate.middleware.js';

const router = Router();

router.post(
  '/register',
  [
    body('name').notEmpty().withMessage('Name is required'),
    body('email').isEmail().withMessage('Valid email is required'),
    body('password').isLength({ min: 6 }).withMessage('Password must be at least 6 characters'),
  ],
  validate,
  register
);

router.post(
  '/login',
  [
    body('email').isEmail().withMessage('Valid email is required'),
    body('password').notEmpty().withMessage('Password is required'),
  ],
  validate,
  login
);

router.post('/logout', logout);
router.post('/refresh', refreshToken);
router.get('/me', protect, getMe);

export default router;

File: app/routes/auth_routes.py
from fastapi import APIRouter, Depends
from app.models.auth import RegisterRequest, LoginRequest, RefreshRequest
from app.controllers import auth_controller
from app.middlewares.auth import get_current_user

router = APIRouter(prefix="/auth", tags=["Auth"])


@router.post("/register")
async def register(body: RegisterRequest):
    return await auth_controller.register(body)


@router.post("/login")
async def login(body: LoginRequest):
    return await auth_controller.login(body)


@router.post("/logout")
async def logout():
    return await auth_controller.logout()


@router.post("/refresh")
async def refresh_token(body: RefreshRequest):
    return await auth_controller.refresh_token(body)


@router.get("/me")
async def get_me(current_user=Depends(get_current_user)):
    return await auth_controller.get_me(current_user)
